elementary/monitor/api/tests/tests.py

```diff
@@ -28,34 +52,207 @@ def get_test_sub_type_unique_id(
         return f"{model_unique_id}.{test_unique_id}.{column_name if column_name else 'None'}.{test_sub_type if test_sub_type else 'None'}"
 
     def get_tests_metadata(
-        self, days_back: Optional[int] = 7
+        self,
+        days_back: Optional[int] = 7,
+        invocation_id: str = None,
+        should_cache: bool = True,
     ) -> List[TestMetadataSchema]:
         run_operation_response = self.dbt_runner.run_operation(
-            macro_name="get_test_results", macro_args=dict(days_back=days_back)
+            macro_name="get_test_results",
+            macro_args=dict(days_back=days_back, invocation_id=invocation_id),
         )
         tests_metadata = (
             json.loads(run_operation_response[0]) if run_operation_response else []
         )
-        return [TestMetadataSchema(**test_metadata) for test_metadata in tests_metadata]
+        tests_metadata = [
+            TestMetadataSchema(**test_metadata) for test_metadata in tests_metadata
+        ]
+        if should_cache:
+            self.set_run_cache(key=TESTS_METADATA, value=tests_metadata)
+        return tests_metadata
+
+    def _get_invocation_from_filter(
+        self, filter: DataMonitoringFilter
+    ) -> Optional[DbtInvocationSchema]:
+        invocation = DbtInvocationSchema()
+        if filter.invocation_id:
+            invocation = self.invocatons_api.get_invocation_by_id(
+                type="test", invocation_id=filter.invocation_id
+            )
+        elif filter.invocation_time:
+            invocation = self.invocatons_api.get_invocation_by_time(
+                type="test", invocation_max_time=filter.invocation_time
+            )
+        elif filter.last_invocation:
+            invocation = self.invocatons_api.get_last_invocation(type="test")
```

Comment:
If none of these filters exists, it means there's no filter by invocation?
(if that's the case, worth adding a comment that clarifies it)